File: JavaCivitas/src/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa{

    private int valor;

    SorpresaPorCasaHotel(int valor_a, String texto){

        super(texto);
        valor = valor_a;

    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){

        if (jugadorCorrecto(actual, todos) && todos.get(actual) != null){
            super.informe(actual, todos);
            todos.get(actual).modificarSaldo(valor * todos.get(actual).cantidadCasasHoteles());
        }

    }

}

File: civitasruby/lib/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class SorpresaPorCasaHotel < Sorpresa
  def initialize(texto,valor)
    super(texto)
    @valor = valor
  end
  
  def aplicar_a_jugador(actual,todos)
    if(jugador_correcto(actual,todos))
        super(actual,todos)
        todos[actual].modificar_saldo(@valor*todos[actual].cantidad_casas_hoteles)
    end
  end
  
  def to_string
    a = "Por Casa-Hotel"
    return a
  end
  public_class_method :new
end
end